spiflash: build fails to link

Compiling spiflash on Ubuntu 16.04 or 18.04 fails with the following error. I see the same locally and in CI.

The libraries are actually installed into /usr/lib/x86_64-linux-gnu and they export the asked-for symbols.

```
philipp@philipp-VirtualBox:~/src/opentitan$ make -C sw/host/spiflash clean all
make: Entering directory '/home/philipp/src/opentitan/sw/host/spiflash'
rm -f spiflash *.o ../vendor/mpsse/mpsse.o ../vendor/mpsse/support.o
g++ -I../../..  -Wall -std=c++14 -c -o verilator_spi_interface.o verilator_spi_interface.cc
g++ -I../../..  -Wall -std=c++14 -c -o updater.o updater.cc
g++ -I../../..  -Wall -std=c++14 -c -o ftdi_spi_interface.o ftdi_spi_interface.cc
cc -I../../..  -std=gnu99 -c -o ../vendor/mpsse/mpsse.o ../vendor/mpsse/mpsse.c
cc -I../../..  -std=gnu99 -c -o ../vendor/mpsse/support.o ../vendor/mpsse/support.c
g++ -I../../..  -Wall -std=c++14 -lcrypto -lftdi1 -lusb-1.0 -L/usr/lib/x86_64-linux-gnu spiflash.cc -o spiflash spi_interface.h updater.h verilator_spi_interface.h ftdi_spi_interface.h verilator_spi_interface.o updater.o ftdi_spi_interface.o ../vendor/mpsse/mpsse.o ../vendor/mpsse/support.o
updater.o: In function `opentitan::spiflash::(anonymous namespace)::HashFrame(opentitan::spiflash::Frame*)':
updater.cc:(.text+0x172): undefined reference to `SHA256_Init'
updater.cc:(.text+0x191): undefined reference to `SHA256_Update'
updater.cc:(.text+0x1b0): undefined reference to `SHA256_Update'
updater.cc:(.text+0x1dc): undefined reference to `SHA256_Update'
updater.cc:(.text+0x1f2): undefined reference to `SHA256_Final'
updater.o: In function `opentitan::spiflash::Updater::Run()':
updater.cc:(.text+0x66a): undefined reference to `SHA256'
../vendor/mpsse/mpsse.o: In function `OpenIndex':
mpsse.c:(.text+0x19c): undefined reference to `ftdi_init'
mpsse.c:(.text+0x1cb): undefined reference to `ftdi_set_interface'
mpsse.c:(.text+0x1f1): undefined reference to `ftdi_usb_open_desc_index'
mpsse.c:(.text+0x271): undefined reference to `ftdi_usb_reset'
mpsse.c:(.text+0x289): undefined reference to `ftdi_set_latency_timer'
mpsse.c:(.text+0x2a1): undefined reference to `ftdi_write_data_set_chunksize'
mpsse.c:(.text+0x2b9): undefined reference to `ftdi_read_data_set_chunksize'
mpsse.c:(.text+0x2d6): undefined reference to `ftdi_set_bitmode'
mpsse.c:(.text+0x31a): undefined reference to `ftdi_set_bitmode'
mpsse.c:(.text+0x369): undefined reference to `ftdi_usb_purge_buffers'
mpsse.c:(.text+0x385): undefined reference to `ftdi_set_bitmode'
../vendor/mpsse/mpsse.o: In function `Close':
mpsse.c:(.text+0x400): undefined reference to `ftdi_set_bitmode'
mpsse.c:(.text+0x410): undefined reference to `ftdi_usb_close'
mpsse.c:(.text+0x420): undefined reference to `ftdi_deinit'
../vendor/mpsse/mpsse.o: In function `ErrorString':
mpsse.c:(.text+0xbfd): undefined reference to `ftdi_get_error_string'
../vendor/mpsse/mpsse.o: In function `SetDirection':
mpsse.c:(.text+0x18ed): undefined reference to `ftdi_set_bitmode'
../vendor/mpsse/mpsse.o: In function `WritePins':
mpsse.c:(.text+0x194d): undefined reference to `ftdi_write_data'
../vendor/mpsse/mpsse.o: In function `ReadPins':
mpsse.c:(.text+0x19a3): undefined reference to `ftdi_read_pins'
../vendor/mpsse/support.o: In function `raw_write':
support.c:(.text+0x3b): undefined reference to `ftdi_write_data'
../vendor/mpsse/support.o: In function `raw_read':
support.c:(.text+0x94): undefined reference to `ftdi_read_data'
support.c:(.text+0xcc): undefined reference to `ftdi_usb_purge_rx_buffer'
collect2: error: ld returned 1 exit status
Makefile:27: recipe for target 'spiflash' failed
make: *** [spiflash] Error 1
make: Leaving directory '/home/philipp/src/opentitan/sw/host/spiflash'
```

--- a/sw/host/spiflash/Makefile
+++ b/sw/host/spiflash/Makefile
@@ -24,7 +24,7 @@ all: $(PROGRAM)
 	$(CXX) $(CXXFLAGS) -c -o $@ $<
 
 $(PROGRAM): $(PROGRAM).cc $(OBJS) $(MPSSE_OBJS)
-	$(CXX) $(CXXFLAGS) $(LDFLAGS) spiflash.cc -o spiflash $(DEPS) $(OBJS) $(MPSSE_OBJS)
+	$(CXX) $(CXXFLAGS) spiflash.cc -o spiflash $(DEPS) $(OBJS) $(MPSSE_OBJS) $(LDFLAGS)
 
 clean:
 	rm -f $(PROGRAM) *.o ${MPSSE_OBJS}

